### PR TITLE
Moved extending documentation and cleaned up a bit

### PR DESCRIPTION
**Pull request by wburns**

### CODE REVIEW

File: documentation/src/main/asciidoc/user_guide/extending.adoc [hunk 1]
```diff
@@ -1,26 +1,29 @@
-= Extending Infinispan
-The Infinispan community
-:icons: font
+== Extending Infinispan
+Infinispan can be extended to provide the ability for an end user to provide their own
+custom commands that can be used which
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/user_guide/extending.adoc]: Doesn't make any sense :)

File: documentation/src/main/asciidoc/user_guide/extending.adoc [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
+== Extending Infinispan
+Infinispan can be extended to provide the ability for an end user to provide their own
+custom commands that can be used which 
 
-This guide is intended for anyone extending or customizing Infinispan.
+=== Custom Commands
+Infinispan makes use of a link:$$http://en.wikipedia.org/wiki/Command_pattern$$[command/visitor pattern] to
+implement the various top-level methods you see on the public-facing API.
+This is explained in further detail in the
+link:$$http://community.jboss.org/wiki/ArchitecturalOverview$$[Architectural Overview] on the Infinispan
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/user_guide/extending.adoc]: We really need to move those old docs into this documentation.

File: documentation/src/main/asciidoc/user_guide/extending.adoc (hunk 1 above)
Comment (wburns) [inline documentation/src/main/asciidoc/user_guide/extending.adoc]: Thanks, missed finishing it :)

File: documentation/src/main/asciidoc/user_guide/extending.adoc (hunk 2 above)
Comment (wburns) [inline documentation/src/main/asciidoc/user_guide/extending.adoc]: Added a new adoc.

**Comment by wburns** [comment]
Updated from review

**Comment by slaskawi** [comment]
Looking

**Comment by slaskawi** [comment]
Integrated, thanks @wburns!

**Pull request merged**